src/coreclr/inc/regdisp.h

```diff
@@ -398,18 +394,8 @@ inline void SyncRegDisplayToCurrentContext(REGDISPLAY* pRD)
 {
     LIMITED_METHOD_CONTRACT;
 
-#if defined(TARGET_64BIT)
-    pRD->SP         = (INT_PTR)GetSP(pRD->pCurrentContext);
-    pRD->ControlPC  = (INT_PTR)GetIP(pRD->pCurrentContext);
-#elif defined(TARGET_ARM)
-    pRD->SP         = (DWORD)GetSP(pRD->pCurrentContext);
-    pRD->ControlPC  = (DWORD)GetIP(pRD->pCurrentContext);
-#elif defined(TARGET_X86)
-    pRD->SP         = (DWORD)GetSP(pRD->pCurrentContext);
-    pRD->ControlPC  = (DWORD)GetIP(pRD->pCurrentContext);
-#else // TARGET_X86
-    PORTABILITY_ASSERT("SyncRegDisplayToCurrentContext");
-#endif
+    pRD->SP         = (TADDR)GetSP(pRD->pCurrentContext);
+    pRD->ControlPC  = (TADDR)GetIP(pRD->pCurrentContext);
```

Comment:
```suggestion
    pRD->SP         = GetSP(pRD->pCurrentContext);
    pRD->ControlPC  = GetIP(pRD->pCurrentContext);
```
Let's try delete the cast and see whether it compiles everywhere